catalog/core/catalog-core-directorymonitor/src/test/java/org/codice/ddf/catalog/content/monitor/osgi/CdmServiceWaiterTest.java

```
@@ -0,0 +1,154 @@
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableSet;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.codice.ddf.catalog.content.monitor.CdmState;
+import org.junit.Before;
+import org.junit.Test;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+
+public class CdmServiceWaiterTest {
+
+  private static final String TEST_PROPERTY_ID = "propertyId";
+
+  private static final String TEST_CLASS_NAME = "className";
+
+  private static final String TEST_PROPERTY_VALUE_1 = "propertyValue1";
+
+  private static final String TEST_PROPERTY_VALUE_2 = "propertyValue2";
+
+  private ScheduledExecutorService executorService;
+
+  @Before
+  public void setup() {
+    setupExecutor();
+  }
+
+  @Test
+  public void testWaitForPropertyValues() {
+    CdmState cdmState = new CdmState();
+    CdmServiceWaiter cdmServiceWaiter =
+        setupServiceWaiter(cdmState, 5, TEST_PROPERTY_VALUE_1, TEST_PROPERTY_VALUE_2);
+
+    try {
+      cdmServiceWaiter.waitFor(ImmutableSet.of(TEST_PROPERTY_VALUE_1, TEST_PROPERTY_VALUE_2));
+    } catch (InterruptedException e) {
+      fail("Should not be interrupted");
+    }
+
+    verify(executorService, times(1)).shutdownNow();
+    assertThat(cdmState.isInputTransformersReady(), is(true));
+  }
+
+  @Test
+  public void testWaitForPropertyValuesTimeout() {
+    CdmState cdmState = new CdmState();
+    CdmServiceWaiter cdmServiceWaiter = setupServiceWaiter(cdmState, 5, TEST_PROPERTY_VALUE_1);
+
+    try {
+      cdmServiceWaiter.waitFor(ImmutableSet.of("doesntexist"));
+    } catch (InterruptedException e) {
+      fail("Should not be interrupted");
```

✏️ Just let it bubble out of the test and fail the test that way